src/rust-crypto/backup.ts

```diff
@@ -205,22 +205,41 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
      */
     public async importBackedUpRoomKeys(keys: IMegolmSessionData[], opts?: ImportRoomKeysOpts): Promise<void> {
         const keysByRoom: Map<RustSdkCryptoJs.RoomId, Map<string, IMegolmSessionData>> = new Map();
+        let invalid = 0;
         for (const key of keys) {
+            if (typeof key.room_id != "string" || typeof key.session_id != "string") {
+                // if key.room_id or key.session_id are invalid, the wasm
+                // bindings will fail when decoding keysByRoom
+                invalid++;
+                const importOpt: ImportRoomKeyProgressData = {
+                    total: keys.length,
+                    successes: 0,
+                    stage: "load_keys",
+                    failures: invalid,
+                };
+                opts?.progressCallback?.(importOpt);
```

Comment:
since this loop is synchronous, I'm not convinced we should do this callback on each iteration.